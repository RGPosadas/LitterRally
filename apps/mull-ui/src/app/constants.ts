export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
