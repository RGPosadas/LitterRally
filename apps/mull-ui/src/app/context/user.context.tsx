import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
