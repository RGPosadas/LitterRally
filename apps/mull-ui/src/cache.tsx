import { InMemoryCache, ReactiveVar, makeVar } from '@apollo/client';

export const cache: InMemoryCache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        login: {
          read() {
            return isLoggedIn();
          },
        },
      },
    },
  },
});

const loginInitialValue: boolean = false;

export const isLoggedIn: ReactiveVar<boolean> = makeVar<boolean>(loginInitialValue);
