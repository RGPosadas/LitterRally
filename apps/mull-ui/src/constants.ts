import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
