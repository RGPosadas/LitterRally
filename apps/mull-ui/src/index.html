<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Müll</title>
    <base href="/" />

    <link rel="icon" type="image/x-icon" href="favicon.png" />
    <link rel="manifest" href="manifest.json" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="description" content="Müll: Help cleanup the world!" />
    <meta name="robots" content="index, follow" />
    <meta name="theme-color" content="#27b09a" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="default" />
    <meta name="apple-mobile-web-app-title" content="my-app" />
    <link rel="apple-touch-icon" href="assets/icons/apple-touch-icon.png" />
  </head>
  <body>
    <div id="root"></div>
    <noscript>
      <div style="font-size: 20px; display: block">
        <div>Please enable JavaScript to continue using this application.</div>
        <div>
          <a href="https://www.enable-javascript.com">https://www.enable-javascript.com</a>
        </div>
      </div>
    </noscript>
  </body>
</html>
