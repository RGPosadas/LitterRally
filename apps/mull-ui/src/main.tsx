import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
import { createUploadLink } from 'apollo-upload-client';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import App from './app/app';
import { environment } from './environments/environment';

if ('serviceWorker' in navigator) {
  window.addEventListener('load', function () {
    navigator.serviceWorker.register('/service-worker.js').then(
      function (registration) {
        // Registration was successful
        console.log('ServiceWorker registration successful with scope: ', registration.scope);
      },
      function (err) {
        // registration failed :(
        console.log('ServiceWorker registration failed: ', err);
      }
    );
  });
}

const client = new ApolloClient({
  link: createUploadLink({
    uri: `${environment.backendUrl}/graphql`,
  }),
  cache: new InMemoryCache(),
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);
