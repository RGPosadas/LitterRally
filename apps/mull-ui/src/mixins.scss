@mixin vertical-center-fixed {
  top: 50%;
  transform: translateY(-50%);
}

@mixin horizontal-center-fixed {
  left: 50%;
  transform: translateX(-50%);
}
