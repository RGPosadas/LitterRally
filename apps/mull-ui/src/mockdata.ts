import { DetectionResult, EventRestriction, ISerializedEvent, ISerializedPost } from '@mull/types';

// TODO delete when US 2.1 is completed
export const dummyProfilePictures: string[] = [
  'https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg',
  'https://i.pinimg.com/236x/92/f0/ed/92f0edd9b0ecefdd5b7a48b8e1f7d340.jpg',
  'https://pbs.twimg.com/profile_images/748593045566853124/9DDVz0uT_400x400.jpg',
];

export const dummyEvent: ISerializedEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: '2009-06-15T13:45:30',
  endDate: '2009-06-15T13:45:30',
  location: {
    id: 1,
    title: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      id: 1,
      name: 'User 1',
    },
    {
      id: 2,
      name: 'User 2',
    },
  ],
  host: {
    id: 3,
    name: 'Test user',
  },
  coHosts: [
    {
      id: 4,
      name: 'First CoHost',
    },
  ],
  image: {
    id: 1,
    mediaType: 'jpeg',
  },
};

export const dummyDetectionResults: DetectionResult[] = [
  {
    bndBox: {
      height: 50,
      width: 100,
      x: 20,
      y: 40,
    },
    class: 'bottle',
    confidence: 0.8,
  },
];

const mockUser1 = {
  id: 1,
  password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
  email: 'mock@mock.com',
  timezone: '',
  name: 'Bob',
  dob: new Date(),
  description: 'I am very good programmer',
  avatar: { id: 1, mediaType: 'png' },
  friends: null,
  tokenVersion: 0,
  registrationMethod: null,
  joinDate: new Date('2020-10-31T22:09:00.000Z'),
};

const mockUser2 = {
  id: 2,
  password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
  email: 'mock@mock.com',
  timezone: '',
  name: 'Bob',
  dob: new Date(),
  description: 'I am very good programmer',
  avatar: { id: 1, mediaType: 'png' },
  friends: null,
  tokenVersion: 0,
  registrationMethod: null,
  joinDate: new Date('2020-10-31T22:09:00.000Z'),
};

const mockChannel = {
  id: 3,
  name: 'announcements',
  rights: 0,
  posts: null,
  participants: null,
};

export const mockPosts: ISerializedPost[] = [
  {
    parentPost: null,
    id: 1,
    channel: mockChannel,
    user: mockUser1,
    message: 'message 1 from user 1, current user id is 1',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 2,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 2 from user 2',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 3,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 3 from user 3',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 4,
    channel: mockChannel,
    user: mockUser2,
    message: '',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: {
      id: 1,
      mediaType: 'jpeg',
    },
  },
  {
    parentPost: null,
    id: 5,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 4 from user 4 with a media',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: {
      id: 1,
      mediaType: 'png',
    },
  },
];

export const mockChannelWithPosts = {
  id: 3,
  name: 'announcements',
  rights: 0,
  posts: mockPosts,
  event: dummyEvent,
};
