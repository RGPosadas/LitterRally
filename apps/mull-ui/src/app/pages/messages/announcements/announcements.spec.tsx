import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import { ROUTES } from '../../../../../src/constants';
import { UserProvider } from '../../../context/user.context';
import AnnouncementsPage from './announcements';

describe('AnnouncementsPage', () => {
  const renderHelper = (history) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <MockedProvider>
          <Router history={history}>
            <AnnouncementsPage />
          </Router>
        </MockedProvider>
      </UserProvider>
    );
  };

  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.MY_EVENTS.url);

    const { baseElement } = render(renderHelper(history));
    expect(baseElement).toBeTruthy();
  });
});
