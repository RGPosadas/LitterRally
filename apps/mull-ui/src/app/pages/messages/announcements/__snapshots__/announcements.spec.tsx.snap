// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Announcements should match snapshot 1`] = `
<div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="other-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Hello, and welcome!
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Filler text is text that shares some characteristics of a real written text, but is random or otherwise generated. It may be used to display a sample of fonts, generate text for testing, or to spoof an e-mail spam filter.
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
      </p>
    </div>
  </div>
  <div
    className="chat-container"
  >
    <p
      className="announcement-time"
    >
      13:00 21/10/20
    </p>
    <div
      className="current-user-chat-container"
    >
      <img
        alt="user"
        className="user-picture"
        src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
      />
      <p
        className="chat-bubble"
      >
        Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.
      </p>
    </div>
  </div>
  <form
    className="chat-input-container"
  >
    <div
      className="custom-file-upload-container"
    >
      <label
        className=""
        htmlFor="imageFile"
      >
        <div
          className="file-upload-feedback"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-images fa-w-18 custom-file-upload-icon"
            data-icon="images"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480 416v16c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V176c0-26.51 21.49-48 48-48h16v208c0 44.112 35.888 80 80 80h336zm96-80V80c0-26.51-21.49-48-48-48H144c-26.51 0-48 21.49-48 48v256c0 26.51 21.49 48 48 48h384c26.51 0 48-21.49 48-48zM256 128c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-96 144l55.515-55.515c4.686-4.686 12.284-4.686 16.971 0L272 256l135.515-135.515c4.686-4.686 12.284-4.686 16.971 0L512 208v112H160v-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </label>
      <input
        accept="image/*"
        className="custom-file-upload-input"
        data-testid="file"
        id="imageFile"
        onChange={null}
        type="file"
      />
    </div>
    <div
      className="custom-text-input-container "
    >
      <label
        className="custom-text-input-label"
        htmlFor="description"
      >
        
      </label>
      <input
        className="custom-text-input input-border "
        data-testid="custom-text-input"
        id="description"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="chat-input-button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-paper-plane fa-w-16 send-icon"
          data-icon="paper-plane"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
